lib/galaxy/tool_util/parameters/case.py

```diff
@@ -249,8 +253,26 @@ def _merge_into_state(
     else:
         test_input = _input_for(state_path, inputs)
         if test_input is not None:
+            input_value: Any
             if isinstance(tool_input, (DataCollectionParameterModel,)):
-                input_value = test_input.get("attributes", {}).get("collection")
+                input_value = TestCollectionDef.from_dict(
+                    cast(XmlTestCollectionDefDict, test_input.get("attributes", {}).get("collection"))
+                ).test_format_to_dict()
+            elif isinstance(tool_input, (DataParameterModel,)):
+                data_tool_input = cast(DataParameterModel, tool_input)
```

Comment:
Is this cast necessary ? mypy has no problems with this

Reply:
I did not know that - good to know. I think I've gotten in the habit of doing that with parameters because I so often am dispatching on the parameter_type string maybe. I think it kind of reads better this way but I will admit it is more code. I will defer to you and remove the casts in my working branch if you'd like.